content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+
+**Bevy 0.8** adds public "untyped" ECS apis that enable retrieving [lifetimed pointers](#ecs-lifetimed-pointers) to component and resource data using [`ComponentId`] instead of actual Rust types. 
+
+```rust
+let health_ptr: Ptr = world.entity(player).get_by_id(heath_component_id).unwrap();
+```
+
+These, when combined with our Reflection apis, provide the tools needed to start building scripting support!
+
+`@jakobhellermann` has [started building their own JavaScript / TypeScript plugin for Bevy](https://github.com/jakobhellermann/bevy_mod_js_scripting/blob/main/assets/scripts/debug.ts). Note that:
+1. This plugin is still very much a work in progress and is not ready to be used in projects.
+2. This is an unofficial community effort. Bevy will not be adding official JavaScript/TypeScript support.
+
+Here is a TypeScript snippet from their repository that queries Bevy ECS data from the script:
+
+```typescript
+const ballQuery = world.query({
+    components: [ballId, transformId, velocityId],
+});
+for (const item of ballQuery) {
+    let [ball, transform, velocity] = item.components;
+    velocity = velocity[0];
+
+    info(velocity.toString());
+}
+```
+
+[`ComponentId`]: https://docs.rs/bevy/0.8.0/bevy/ecs/component/struct.ComponentId.html
+
+## ECS "lifetimed pointers"
+
+<div class="release-feature-authors">authors: @TheRawMeatball, @BoxyUwU, @jakobhellermann</div>
+
+Bevy ECS has been refactored to use lifetimed, type-erased pointers instead of "raw pointers", which significantly improved the safety and legibility of our internals without compromising performance or flexibility.
+
+From a high level, this enables us to "retain" the lifetime of World borrows throughout our internals while still using type-erased APIs to support scenarios like [3rd party scripting languages](#scripting-modding-progress-untyped-ecs-apis).
+
+By retaining this lifetime, we can rely more on Rust's borrow checker to yell at us when we are doing something unsound. And, as it happens, this caught a number of soundness bugs!
+
+## ECS Query Internals Refactors
+
+<div class="release-feature-authors">authors: @BoxyUwU</div>
+
+`@BoxyUwU` has been hard at work refactoring our [`Query`] internals to be simpler and easier to read:
+
+* `ReadOnlyFetch` was replaced with [`ReadOnlyWorldQuery`], moving this trait constraint "up a level", making it easier to express in the type system.
+* "QF Fetch generics" were removed from [`Query`] and [`QueryState`] methods and types in favor of [`WorldQuery`] usage, making filters and normal fetches consistent in the type system and simpler to express.
+
+We have more changes in this vein planned for the next release. Bevy ECS internals are starting to be more approachable!
```

Comment:
Phrasing of second sentence is awkward; suggestions on how to fix welcome.

Reply:
```suggestion
We have more changes in this vein planned for the next release. Bevy ECS internals are becoming considerably easier to understand!
```